src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -913,10 +914,21 @@ adios2::Mode ADIOS2IOHandlerImpl::adios2AccessMode( )
     case Access::READ_ONLY:
         return adios2::Mode::Read;
     case Access::READ_WRITE:
-        std::cerr << "ADIOS2 does currently not yet implement ReadWrite "
-                     "(Append) mode. "
-                  << "Replacing with Read mode." << std::endl;
-        return adios2::Mode::Read;
+        if( auxiliary::directory_exists( fullPath ) ||
+            auxiliary::file_exists( fullPath ) )
+        {
+            std::cerr << "ADIOS2 does currently not yet implement ReadWrite "
+                         "(Append) mode. "
+                      << "Replacing with Read mode." << std::endl;
+            return adios2::Mode::Read;
+        }
+        else
+        {
+            std::cerr << "ADIOS2 does currently not yet implement ReadWrite "
+                         "(Append) mode. "
+                      << "Replacing with Write mode." << std::endl;
+            return adios2::Mode::Write;
```

Comment:
Is this safe to keep going instead of asking for another mode?

Reply:
Choosing read mode is safe anyway and write mode is only chosen if the file does not exist, so it should be safe, too. I can imagine that there might be issues in parallel, I should test how ADIOS2 behaves for that and potentially add an MPI barrier.

Reply:
According to some quick experiments I did, ADIOS does not create files on the disk if not all of the parallel writers have called `Engine::open()` yet. So we could leave it as it is or alternatively add an MPI Barrier for some extra safety.